src/main/java/org/apache/solr/handler/dataimport/SolrPaths.java

```diff
@@ -0,0 +1,78 @@
+package org.apache.solr.handler.dataimport;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.naming.NoInitialContextException;
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Code copied from Solr 8.11.1
+ */
+public class SolrPaths {
```

Comment:
note: `locateSolrHome` method is used in `SimplePropertiesWriter` class. It was removed from Solr 9, so I copied it here from 8.11.1.